engine/storage/datamotion/src/main/java/org/apache/cloudstack/storage/motion/StorageSystemDataMotionStrategy.java

```diff
@@ -253,18 +262,64 @@ private boolean canHandle(DataObject dataObject) {
         return false;
     }
 
+    /**
+     * True if volumes source storage are NFS
+     */
```

Comment:
@nvazquez The below method returns true when the first NFS pool of a volume found in the given volumes list. Is that the intent? or it has to check all the volumes with NFS pool?

Reply:
Yes.

Reply:
@rhyd Ok.

Reply:
@sureshanaparti yes, that was the intent as it was planned to be for live storage migrations from host H1 to host H2 and storage A to storage B, i.e. multiple volumes on the same storage so no need to check all of them. But it can be extended to check all the volumes in the map

Reply:
@nvazquez If multiple volumes are on the same storage is what being checked, why Map with multiple volumes & looping around them is required?

it can be extended? any current use case supports this? if extended, it only checks for NFS pool.